src/apof/portal/pipeline.py

```diff
@@ -0,0 +1,8 @@
+def get_avatar(backend, strategy, response, user=None, *args, **kwargs):
+    url = None
+    if backend.name == 'google-oauth2':
+        url = response['image'].get('url')
+    if url:
+        prof = user.profile
+        prof.avatar = url
+        user.save()
```

Comment:
There shouldn't be, since anonymous users doesn't have acces to any app feature except logging page which doesn't require any user profile actions. But for precaution, it can be closed between try catch block.